src/trio/_core/_tests/test_ki.py

```diff
@@ -331,7 +334,8 @@ def kill_during_shutdown() -> None:
 
         token.run_sync_soon(kill_during_shutdown)
 
-    with pytest.raises(KeyboardInterrupt):
+    # Does not wrap in an ExceptionGroup(!!)
+    with testing.raises(KeyboardInterrupt):
```

Comment:
I would love an explanation in a comment for both this and the subsequent one. I think I figured out why ("if it's from a sync function, then you cannot have code running concurrently") and hopefully that's right?

Reply:
I don't know! (and haven't looked into them yet)

And regardless we might consider whether it's the best interface to have them differ, or if inner nurseries should be hidden in some cases (either by reraising, or running stuff with `strict_exception_groups=False`)

Reply:
Looked into it:
1. `check_unprotected_kill` opens a nursery, and only inside of it does it call `raiser`, so it will be wrapped in an exceptiongroup from the nursery.
2. `check_protected_kill` does the same.
3. `check_kill_during_shutdown` does not touch any nurseries, so there's nothing that would wrap exceptions
4. `main_1` likewise does not do anything with nurseries.

So it seems very straightforward, will update/add comments.